packages/core/src/components/workflow-steps/workflow-steps.tsx

```diff
@@ -65,6 +65,50 @@ export class WorkflowSteps {
     });
   }
 
+  private observer: MutationObserver;
+
+  componentDidLoad() {
+    let steps = this.getSteps();
+
+    let mList = document.getElementById('stepList'),
+      options = {
+        childList: true,
+      };
+    this.observer = new MutationObserver(callback);
+
+    function callback(mutations) {
+      for (let mutation of mutations) {
+        steps.push(mutation.addedNodes[0]);
+
+        if (mutation.type === 'childList') {
+          steps.forEach((element, index) => {
```

Comment:
Please avoid redundant code (here and between `componentDidRender()`) use a function